met/src/tools/other/pb2nc/pb2nc.cc

```diff
@@ -1513,38 +1526,45 @@ void process_pbfile(int i_pb) {
 
          // Derive quantities which can be derived from
          // P, Q, T, Z, U, V
-         for(i=0; i<n_derive_gc; i++) {
-
-            bufr_var_index = bufr_derive_code[i];
-            if(keep_bufr_obs_index(bufr_var_index)) {
-
-               // Only derive PRMSL for surface message
-               if(derive_gc[i] == prmsl_grib_code &&
-                  !conf_info.surface_message_types.has(hdr_typ))
-                  continue;
-
-               // Store the grib code to be derived
-               obs_arr[1] = (float)bufr_var_index;
-
-               // Derive the value for the grib code
-               obs_arr[4] = derive_grib_code(derive_gc[i], pqtzuv,
-                                             pqtzuv_qty, hdr_lat,
-                                             quality_mark);
-
-               if(is_eq(obs_arr[4], fill_value)) continue;
-
-               addObservation(obs_arr, (string)hdr_typ, (string)hdr_sid, hdr_vld_ut,
-                     hdr_lat, hdr_lon, hdr_elv, quality_mark,
-                     OBS_BUFFER_SIZE);
-
-               // Increment the current and total observations counts
-               n_file_obs++;
-               n_total_obs++;
-
-               // Increment the number of obs counter for this header
-               n_hdr_obs++;
-            }
-         } // end for i
+         if (n_derive_gc > bufr_derive_cfgs.size()) {
+            mlog << Warning << "\n" << method_name << " -> "
```

Comment:
Recommend changing "Warning" to "Debug(3)".  Not requesting any derivations shouldn't be a warning, just an informational message.

Reply:
I actually don't understand this logic.  n_derive_gc = 6.  If I configure pb2nc to only derive DPT, for example, what is bufr_derive_cfgs.size()?